drivers/storage/portworx/component/security.go

```diff
@@ -324,6 +372,70 @@ func (c *security) deleteSecret(
 	return k8sutil.DeleteSecret(c.k8sClient, name, cluster.Namespace, *ownerRef)
 }
 
+// closeSdkConn closes the sdk connection and resets it to nil
+func (c *security) closeSdkConn() {
+	if c.sdkConn == nil {
+		return
+	}
+
+	if err := c.sdkConn.Close(); err != nil {
+		logrus.Errorf("failed to close sdk connection: %s", err.Error())
+	}
+	c.sdkConn = nil
+}
+
+func (c *security) updateSystemGuestRole(cluster *corev1alpha1.StorageCluster) error {
+	if *cluster.Spec.Security.Auth.GuestAccess != corev1alpha1.GuestRoleEnabled &&
+		*cluster.Spec.Security.Auth.GuestAccess != corev1alpha1.GuestRoleDisabled &&
+		*cluster.Spec.Security.Auth.GuestAccess != corev1alpha1.GuestRoleManaged {
+		return fmt.Errorf("invalid guest access type: %s", *cluster.Spec.Security.Auth.GuestAccess)
+	}
+
+	// managed, do not interfere with system.guest role
+	if *cluster.Spec.Security.Auth.GuestAccess == corev1alpha1.GuestRoleManaged {
+		return nil
+	}
+
+	var err error
+	c.sdkConn, err = pxutil.GetPortworxConn(c.sdkConn, c.k8sClient, cluster.Namespace)
```

Comment:
In this method, the default grpcserver connection waits for max 1 min for the connection to be ready. That means during the initial install, the operator will get stuck for a 1 min and not proceed with other component installation. That's because the SDK server is not yet up.
So far it was fine because the only thing dependent on SDK server was status update, which is not critical compared to deploying other components.

Can we add a new method on openstorage to take connection timeout from the client and use that in GetPortworxConn? This is out of the scope of this PR, but we will have to do it in this release.

https://github.com/libopenstorage/openstorage/blob/master/pkg/grpcserver/grpcutil.go#L59

Reply:
Created an issue for this.